### PR TITLE
Add a Deploy section and deployment environment settings

### PR DESCRIPTION
**Pull request by emmapons**
Sem-Ver: feature
Signed-off-by: Emma Pons Colnel <emma.pons.colnel@intel.com>

### CODE REVIEW

File: buildbot_travis/ui/src/app/configEditors/projects.tpl.jade [hunk 1]
```diff
@@ -45,4 +45,9 @@ config-page
                         input.form-control(name="repository", type="text", ng-model="project.repository",
                                            placeholder="git://url/to/repo.git", required)
                         p.help-block(ng-if="form.repository.$error.required") The projects's name is required
+                .form-group
+                    label.col-sm-2.control-label deployment environment(s):
+                    .col-sm-10
+                        input-tags(tags="project.deployment_environment", all-tags="allDepEnvs",
```
Comment (tardyp) [inline buildbot_travis/ui/src/app/configEditors/projects.tpl.jade]: I dont see implementation for this allDepsEnv yet

**Comment by tardyp** [comment]
looks good so far..

**Comment by emmapons** [comment]
@bastien31

File: buildbot_travis/ui/src/app/configEditors/bbtravis_config.controller.coffee [hunk 2]
```diff
@@ -31,6 +31,15 @@ class ProjectsConfig extends Controller
                             ret.push(tag)
             return ret
 
+        $scope.allStages = (query) ->
+            ret = []
+            for p in $scope.cfg.projects
+                if p.stages?
+                    for stage in p.stages
+                        if stage.indexOf(query) == 0 and ret.indexOf(stage) < 0
```
Comment (bastien31) [inline buildbot_travis/ui/src/app/configEditors/bbtravis_config.controller.coffee]: Could you add comment to explain what are you checking there?

File: buildbot_travis/ui/src/app/configEditors/input_stage.directive.coffee [hunk 3]
```diff
@@ -0,0 +1,29 @@
+class InputStages extends Directive
```
Comment (bastien31) [inline buildbot_travis/ui/src/app/configEditors/input_stage.directive.coffee]: I guess all this code shall be in input_stage*s*.directive.coffe file? And it is already present (seems similar), so maybe you can just delete that file?

**Comment by tardyp** [comment]
merged in master

**Pull request closed without merging**